### PR TITLE
Add / modify user by template

### PR DESCRIPTION
**Pull request by ImMohammad20000**
now we can add or modify user by **templates** using `POST /api/user/from_template` and `PUT /api/user/{username}`

### CODE REVIEW

File: app/utils/share.py
Comment (M03ED) [inline app/utils/share.py:1]: share.py its a file for making configs for users not creating new user

File: app/models/user.py
Comment (M03ED) [inline app/models/user.py:1]: you can add property here and make user in here at CreateUserFromTemplate class

File: app/views/user.py [hunk 1]
```diff
@@ -6,13 +6,20 @@
...
+    UsersResponse,
+    UserStatus,
+    UserUsagesResponse
+)
+from app.utils import report, share
 
 
 @app.post("/api/user", tags=['User'], response_model=UserResponse)
-def add_user(new_user: UserCreate,
+def add_user(new_user: UserCreate | CreateUserFromTemplate,
```
Comment (M03ED) [inline app/views/user.py:22]: its better to separate api requests

File: requirements.txt [hunk 2]
```diff
@@ -44,6 +44,7 @@ PyMySQL==1.0.3
 pyOpenSSL==23.2.0
 pyparsing==3.0.9
 pypng==0.20220715.0
+PySocks==1.7.1
```
Comment (M03ED) [inline requirements.txt:47]: i can't find where you use this new library

**Review by M03ED — COMMENTED** [review]
i don't think this be approved because templates are useful when they come to dashboard, changes like this is good for when we can set different access for each admin

File: requirements.txt (hunk 2 above)
Comment (ImMohammad20000) [reply requirements.txt:47]: i try to access telegram use socks and that needed this lib so i added

File: app/utils/share.py
Comment (ImMohammad20000) [reply app/utils/share.py:1]: don't want to make routers too long

**Pull request closed without merging**